modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/task/ValidateClientExtensionTask.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.gradle.plugins.workspace.task;
+
+import com.liferay.gradle.plugins.workspace.internal.client.extension.ClientExtension;
+import com.liferay.gradle.plugins.workspace.internal.util.StringUtil;
+import com.liferay.gradle.util.ArrayUtil;
+import com.liferay.petra.string.StringPool;
+
+import java.io.File;
+
+import java.util.Objects;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.Project;
+import org.gradle.api.tasks.TaskAction;
+
+/**
+ * @author Anderson Luiz
+ */
+public class ValidateClientExtensionTask {
+
+	@TaskAction
+	public void validate(ClientExtension clientExtension, Project project) {
+		if (Objects.equals(clientExtension.type, "batch")) {
+			_validateRequiredDirectory(clientExtension, project, "batch");
+			_validateRequiredTypeSettingsKeys(
+				clientExtension, "oAuthApplicationHeadlessServer");
+		}
+
+		if (Objects.equals(clientExtension.type, "instanceSettings")) {
+			_validateRequiredTypeSettingsKeys(clientExtension, "pid");
+		}
+
+		if (Objects.equals(clientExtension.type, "siteInitializer")) {
+			_validateRequiredDirectory(
+				clientExtension, project, "site-initializer");
+			_validateRequiredTypeSettingsKeys(
+				clientExtension, "oAuthApplicationHeadlessServer",
+				"siteExternalReferenceCode", "siteName");
+			_validateTypeSettingsValues(
+				clientExtension, "builtInTemplateType", "site-initializer",
+				"site-template");
+			_validateTypeSettingsValues(
+				clientExtension, "membershipType", "open", "private",
+				"restricted");
+		}
+
+		if (Objects.equals(clientExtension.type, "themeCSS")) {
+			_validateFileExistence(
+				project,
+				clientExtension.typeSettings.get(
+					"frontendTokenDefinition"
+				).toString());
+		}
```

Comment:
This is the only new validation logic, right? Everything else is just being moved around?

Reply:
Yeah